sympy/physics/units/quantities.py

```diff
@@ -114,6 +114,8 @@ def get_dimensional_expr(expr):
             return dim
         elif isinstance(expr, Function):
             args = [Quantity.get_dimensional_expr(arg) for arg in expr.args]
+            if args == [1]:
+                return 1
```

Comment:
We usually use `S.One` or `S(1)`, that is SymPy's integer, which creates fractions under division instead of floating-point numbers.